src/main/java/org/example/requestsAndConnections/GameController.java

```diff
@@ -0,0 +1,59 @@
+package org.example.requestsAndConnections;
+
+import org.example.dto.GameStateDTO;
+import org.example.dto.StepForMoveDTO;
+import org.example.game.Game;
+import org.example.game.Sessions;
+import org.example.myExeptions.MyCustomExceptions;
+
+import java.util.UUID;
+
+public class GameController {
+    static Sessions sessions = new Sessions();
+
+    public UUID newSession() {
+        UUID sessionId = UUID.randomUUID();
+        sessions.addSession(sessionId);
+        return sessionId;
+    }
+
+    public GameStateDTO newGame(UUID sessionId) throws MyCustomExceptions {
+        if (!sessions.containsSession(sessionId)) {
+            throw new MyCustomExceptions("Invalid session id");
+        }
+        return createAndPutNewGameToSessions(sessionId).getGameState();
+    }
+
+    public GameStateDTO getState(UUID sessionId) throws MyCustomExceptions {
+        if (!sessions.containsSession(sessionId)) {
+            throw new MyCustomExceptions("Invalid session id");
+        }
+        if (sessions.hasGames(sessionId)) {
+            Game game = sessions.getLastAddedGame(sessionId);
+            return game.getGameState();
+        } else return newGame(sessionId);
```

Comment:
Such logic can lead to mistakes and unclear design decisions on both FE and the backend. If you think it's necessary to have it - let's rename the function from `get` to `getOrCreate`. But I would separate this logic into two independent functions.